bits.go

```diff
@@ -266,3 +267,50 @@ func fromDeltas(data []byte, buf []uint32) []uint32 {
 	}
 	return buf
 }
+
+// condenseRuneOffsets converts the mostly-predictable runeOffset input
+// into a shorter form based on tuples of the unexpected values.
+// The input is roughly a sequence of y values [100*i for i in range(n)],
+// so we just store (x, y) points where the expectation is violated.
+//
+// This is very rarely a slight pessimization, on repos where there are frequent
+// non-ASCII characters.
+func condenseRuneOffsets(off []uint32) []uint32 {
+	expected := uint32(0)
+	tmp := []uint32{}
+	for x, y := range off {
+		if y != expected {
+			tmp = append(tmp, uint32(x)*runeOffsetFrequency, y)
+			expected = y
+		}
+		expected += runeOffsetFrequency
+	}
+	out := make([]uint32, len(tmp))
+	copy(out, tmp)
+	return out
+}
+
+func runeOffsetLookup(sample []uint32, absR uint32) (uint32, uint32) {
```

Comment:
Can you add some go docs to this function that explain how it works and what the parameters and return arguments are?

Reply:
Done.